tensorflow/core/transforms/remapper/pass.cc

```diff
@@ -306,6 +312,241 @@ class BasePatternActivationRewriter : public BasePatternRewriter {
   }
 };
 
+// NOTE(ezhulenev): See `BatchnormSpatialPersistentEnabled` documentation in the
+// `tensorflow/stream_executor/cuda/cuda_dnn.cc` for details.
+bool BatchnormSpatialPersistentEnabled() {
+#if CUDNN_VERSION >= 7402
+  static bool is_enabled = [] {
+    bool is_enabled = false;
+    TF_CHECK_OK(tensorflow::ReadBoolFromEnvVar(
+        "TF_USE_CUDNN_BATCHNORM_SPATIAL_PERSISTENT",
+        /*default_val=*/false, &is_enabled));
+    return is_enabled;
+  }();
+  return is_enabled;
+#else
+  return false;
+#endif
+}
+
+static std::unique_ptr<OperationState> CreateFusedBatchNormExOpState(
+    OpBuilder &builder, const OpPropertyHelper &helper,
```

Comment:
I would suggest to pass the reference of an OperationState (which can be on the stack instead of dynamic allocation)

Besides, if this is specific to FusedBatchNormExRewriter, it'll be better to be in FusedBatchNormExRewriter scope. Is it?

Reply:
Yes I put this method as in FusedBatchNormExRewriter class, also modified the signature to pass op state as reference.